src/NuGet.Core/NuGet.Packaging.Core/PackageType.cs

```diff
@@ -1,11 +1,14 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
 using NuGet.Shared;
 
 namespace NuGet.Packaging.Core
 {
+    /**
+     * It is important that this type remains immutable due to the cloning of package specs
+     **/
```

Comment:
have you done anything to enforce this? if this is the case, you should not have setters on any properties and all fields should be private

Reply:
It doesn't have any setters, I just put it there, as it's one of the few types that are never cloned.